src/app/components/diary_list.jsx

```diff
@@ -0,0 +1,58 @@
+"use client"
+
+import React from 'react'
+import { useEffect, useState } from 'react'
+import { supabase } from '../lib/supabase';
+import Link from 'next/link';
+
+function DiaryList() {
+  const [entriesData, setEntriesData] = useState();
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        let { data: diary_entries, error } = await supabase
+          .from('diary_entries')
+          .select('*')
+
+        if (error) {
+          throw error;
+        }
+        setEntriesData(diary_entries)
+      } catch (error) {
+      }
+    };
+    fetchData();
+  }, []);
+
+  return (
+    <>
+      <ul role="list" className="divide-y divide-gray-100">
+        {entriesData ?
+          entriesData.map((entry) => (
+            <li key={entry.id} className="flex justify-between gap-x-6 w-screen max-w-7xl px-24 py-5">
```

Comment:
very glad to see the `key` is the `id` not an index. Always best to use an `id` or similar as a key. Nicely done 👍